src/hooks/useTokenContractsConstant.ts

```diff
@@ -0,0 +1,14 @@
+import { Interface } from '@ethersproject/abi'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { useMemo } from 'react'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+
+export function useTokenContractsConstant(tokens: string[], field: 'name' | 'symbol' | 'decimals' | 'totalSupply') {
+  const memoizedTokens = useMemo(() => tokens, [tokens])
+  return useMultipleContractSingleData(memoizedTokens, ERC20Interface, field, undefined, {
+    blocksPerFetch: 10_000,
+  })
+}
```

Comment:
@cmcewen @grabbou 

moving the convo here

this util feels a little weird to me because
- accessing ad-hoc chain data w/ multicall is a pretty normal pattern
- wrapping the array to enforce a possibly-already-memo'd array is memo'd before the hook call when a user could have just written these two lines themselves feels jank

that said, I do like enforcing the very long re-fetch time on a subset of chain data that we can reasonably expect to never change!